docs/HowToUsePeripheral.md

```diff
@@ -501,9 +516,222 @@ static const struct i2s_ops_s g_i2sops = {
 	.i2s_err_cb_register = i2s_err_cb_register,
 };
 ```
+
 ### I2S Initialization and Device Access
-Board shall export ```struct i2s_dev_s* xxx_i2s_initialize(uint16_t port)``` to initialize a i2s port and get the corresponding device structure.    
+Board shall export ```struct i2s_dev_s* xxx_i2s_initialize(uint16_t port)``` to initialize a i2s port and get the corresponding device structure.
 **NOTE**: Current code in s5j_i2s.c has initialize function as ```struct i2s_dev_s* s5j_i2s_initialize(uint16_t port)```.
 
 This i2s initialize function shall be called with the port number to get the corresponding device structure.
 
+
+## PWM
+
+All structure and function prototypes to be implemented by low-level PWM drivers.
+PWM drivers are provided in the header file [pwm.h](../os/include/tinyara/pwm.h).
+
+
+### PWM device driver initialzation
+
+Every PWM channel must implement some function to initialize the PWM device like below.
+Plese see ```imxrt_pwm_register```.
+
+```c
+FAR struct pwm_lowerhalf_s * imxrt_pwm_initialize(int channel)
+{
+	switch (channel) {
+#ifdef CONFIG_IMXRT_PWM1
+	case 1:
+		lower = &g_ch1dev;
+		break;
+	...
+#endif
+#ifdef CONFIG_IMXRT_PWM2
+	case 9:
+		lower = &g_ch9dev;
+		break;
+#endif
+	...
+	if (imxrt_pwm_init(base, submodule, &pwmconfig) == kStatus_Fail) {
+		return NULL;
+	}
+	//...
+	imxrt_pwm_setup((FAR struct pwm_lowerhalf_s *)lower);
+
+	return (FAR struct pwm_lowerhalf_s *)lower;
+}
+
+static void imxrt_pwm_register(int ch)
+{
+#ifdef CONFIG_PWM
+	struct pwm_lowerhalf_s *pwm_ch;
```

Comment:
Actually pwm_ch is not good naming, pwm_lower_half_s doesn't mean channel. it means that pwm's lower device's dev pointer.
So, Here you'd better use *pwm instead of pwm_ch

Reply:
modified.